test/basic/cond_test.c

```diff
@@ -59,6 +59,7 @@ void inc_counter(void *arg)
     ABT_mutex_lock(mutex);
     g_num_incthreads++;
     ABT_cond_wait(broadcast, mutex);
+    ABT_thread_yield();
```

Comment:
what is this `yield` for?

Reply:
To enhance the test.
Without this yield, the original version passes this test.
With this yield, the original does not because of the issue this PR solves.

Reply:
Please create a separate commit for this patch